### ISSUE TITLE
Editor improvements to be on par with VSCode and Codemirror

### ISSUE THREAD

**Issue by ccordoba12** [spyder-ide/spyder#7718]
According to [our users](https://www.reddit.com/r/Python/comments/983etg/spyder_scientific_ide_most_wanted_features/e4erhc9/), we need:

* multi-line editor (PR #4987)
* multi selection with ctrl-d that select the next occurrence of the selected word
* proper dealing of parenthesis when some text is selected - PR #8637.

**Comment by jnsebgosselin** [comment]
Unless you already have something in mind, I can try to finish the multi-line feature I started to work on last year if you wish.

**Comment by ccordoba12** [comment]
That would be really helpful, thanks!

**Comment by bcolsen** [comment]
@ccordoba12 I think #8637 takes care of the third bullet point here

**Comment by athompson673** [comment]
@ccordoba12 I think this can be closed maybe now that multi-cursor has shipped?

**Comment by ccordoba12** [comment]
I think my second bullet point is still unaddressed (unless it's also fixed by the multi-cursors support?)

**Comment by athompson673** [comment]
@ccordoba12 

> unless it's also fixed by the multi-cursors support?

I would consider it partially handled I suppose... I find myself using occurrence highlighting to help in manually placing a cursor at each position using the mouse, which can then be easily used to select with ctrl-shift-right once I have a cursor at each position needed. Between this and find-replace I have personally never missed having a "select next" shortcut. That said, now that the multi-cursor framework is there, it would be possible.

I originally somewhat considered adding some kind of "select next", but ended up not doing it. It brought up a lot of questions for me on how it should best be implemented, as it's a feature I'm not accustomed to using (similar to killring, which I mostly ignored for multicursor). For example do we need a "Skip to next"? Should there be "select all"? "Go to previous"? How do we handle if the next/previous has a cursor present already? Different handling when the primary cursor has a selection vs is simply next to an identifier? How many different shortcuts are we talking about adding? Do we need to do more to visually emphasize the difference between the "primary cursor" and the "extra cursors" to help users keep track?

Once cursor selections are involved, the ability to manually adjust each selection individually is significantly lacking (simple cursors without selections are easy enough to simply remove and replace with the add-remove mouse shortcut. Some degree of adding selections without impacting other cursors is technically possible using the column cursor shortcut as it was designed to somewhat emulate "rectangle select" functionality, but it is not great). 

I can think up a number of different ways it could possibly work, but without a clear direction, I left it for later.

**Comment by ccordoba12** [comment]
> I find myself using occurrence highlighting to help in manually placing a cursor at each position using the mouse, which can then be easily used to select with ctrl-shift-right once I have a cursor at each position needed. Between this and find-replace I have personally never missed having a "select next" shortcut. That said, now that the multi-cursor framework is there, it would be possible.

Yeah, it'd be really cool if you could select a word, wait for occurrences to be highlighted and then add as many cursors as needed to edit them at once. That's what VSCode allows users to do.
